tests/build/build.go

```diff
@@ -123,6 +126,190 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
+					if err != nil {
+						GinkgoWriter.Printf("cannot unmarshal build status: %v\n", err)
+						return false, err
+					}
+
+					return buildStatus.PaC != nil && buildStatus.PaC.State == "enabled", nil
+				}, timeout, interval).Should(BeTrue())
+
+				component, err = f.AsKubeAdmin.HasController.GetHasComponent(annotationsComponentName, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(component.Annotations[build.BuildRequestAnnotationName]).To(BeEmpty(), "request annotation was not deleted")
+			})
+
+			It("build pipeline is triggered after merge", func() {
+				var pipelinerunPullSha string
+				var mergeResult *github.PullRequestMergeResult
+
+				Eventually(func() (bool, error) {
+					prs, err := f.AsKubeAdmin.CommonController.Github.ListPullRequests(helloWorldComponentGitSourceRepoName)
+					if err != nil {
+						GinkgoWriter.Printf("cannot list pull requests: %v\n", err)
+						return false, err
+					}
+
+					for _, pr := range prs {
+						if pr.Head.GetRef() == annotationsBranchName {
+							prNumber = pr.GetNumber()
+							pipelinerunPullSha = pr.Head.GetSHA()
+							return true, nil
+						}
+					}
+					return false, nil
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for init PaC PR to be created")
+
+				mergeResult, err := f.AsKubeAdmin.CommonController.Github.MergePullRequest(helloWorldComponentGitSourceRepoName, prNumber)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Eventually(func() bool {
+					pipelineRunPull, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(annotationsComponentName, applicationName, testNamespace, pipelinerunPullSha)
+					if err != nil {
+						GinkgoWriter.Println("PipelineRun on pull request has not been created yet")
+						return false
+					}
+					pipelineRunPush, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(annotationsComponentName, applicationName, testNamespace, mergeResult.GetSHA())
+					if err != nil {
+						GinkgoWriter.Println("PipelineRun on push has not been created yet")
+						return false
+					}
+					return pipelineRunPull.HasStarted() && pipelineRunPush.HasStarted()
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for the PipelineRuns to start")
+
+				Expect(f.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(testNamespace)).To(Succeed())
+
+				component, err = f.AsKubeAdmin.HasController.GetHasComponent(annotationsComponentName, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(component.Annotations[build.BuildRequestAnnotationName]).To(BeEmpty(), "request annotation was not deleted")
+			})
+
+			It("simple build can be triggered manually in PaC mode", func() {
+				component, err = f.AsKubeAdmin.HasController.GetHasComponent(annotationsComponentName, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				component.Annotations[build.BuildRequestAnnotationName] = build.BuildRequestTriggerSimpleBuildAnnotationValue
```

Comment:
Did you mean to set the value to`build.BuildRequestUnconfigurePaCAnnotationValue`

Reply:
If yes, then we need to check if the purge PR got created removing the pac configuration.

Reply:
Also need to close the Purge PR created by the build-service during cleanup in AfterAll block

Reply:
Good catch about the need to close the purge PR

Reply:
> No, I wanted to test if simple build can be triggered even in PaC mode. I consider PaC unprovision to be covered by your PR #568

the scenario to cover "User moves from PAC to simple build" need to be changed with new annotations as well. That should be done through setting the annotation value to `unconfigure-pac` .